### PR TITLE
Setup build for Java 11 and 16

### PR DESCRIPTION
**Pull request by chrisrohr**
* Project will compile on all versions
* Tests only will be run on 16
* Tests, Jacoco and Sonar will run on 11

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,34 +9,64 @@ jobs:
...
           restore-keys: ${{ runner.os }}-embedmongo
-      - name: Build and analyze
+
+      # Compile the project
+      - name: Build
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+        run: mvn -B -V compile
+
+      # Run tests
```
Comment (sleberknight) [inline .github/workflows/build.yml:59]: Suggest changing the comments to clearly indicate only building on > JDK 11

e.g. something like:

```suggestion
      # Run tests when Java version > 11 (Sonar runs tests and analysis on JDK 11)
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -9,34 +9,64 @@ jobs:
...
+      - name: Run tests
+        if: ${{ env.SONAR_TOKEN == null || env.SONAR_TOKEN == '' || matrix.java_version != '11' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+        run: mvn -B -V verify
+
+      # Run Sonar Analysis
```
Comment (sleberknight) [inline .github/workflows/build.yml:66]: similar to above, suggest something like:

```suggestion
      # Run Sonar Analysis (on Java version 11 only)
```

**Review by sleberknight — APPROVED** [review]
Minor suggestions...apply if you think they make sense. (I was afraid they might be overkill)

**Comment by sleberknight** [comment]
Wait...just checked the build output, and the tests ran on both JDK 11 _and_ 16

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -9,34 +9,64 @@ jobs:
...
+
+      # Compile the project
+      - name: Build
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+        run: mvn -B -V compile
+
+      # Run tests
+      - name: Run tests
+        if: ${{ env.SONAR_TOKEN == null || env.SONAR_TOKEN == '' || matrix.java_version != '11' }}
```
Comment (sleberknight) [inline .github/workflows/build.yml:61]: Is this logic correct? We want to run the tests if _any of_:
1. Sonar token is null
2. Sonar token is empty
3. JDK is 11

Comment (sleberknight) [reply .github/workflows/build.yml:61]: Do we know when and under what situations the Sonar token is null or empty?

Comment (sleberknight) [reply .github/workflows/build.yml:61]: Oh, duh, we set the Sonar token ...but the logic is weird, since why would we need the conditionals?

Comment (chrisrohr) [reply .github/workflows/build.yml:61]: oh those should be &&'s not ||'s

**Pull request merged**